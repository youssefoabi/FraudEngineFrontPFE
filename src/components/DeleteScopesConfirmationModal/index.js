import { bool, func } from 'prop-types';
import { contains, join, pathOr, pickBy, pipe, pluck, values } from 'ramda';
import { Typography } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import React, { useCallback } from 'react';

import { DELETE_SCOPES_REQUEST } from '../ScopesDataGrid/actions';
import ConfirmationModal from '../ConfirmationModal';

export default function DeleteScopesConfirmationModal({ open, toggle }) {
  const dispatch = useDispatch();

  const deleteScopes = useCallback(() => dispatch({ type: DELETE_SCOPES_REQUEST }), [dispatch]);

  const selected = useSelector(pathOr([], ['navigation', 'scopes', 'selected']));
  const namesSelector = pipe(
    pathOr([], ['entities', 'scopes', 'byId']),
    pickBy((value, key) => contains(+key, selected)),
    values,
    pluck('name'),
    join(', '),
  );
  const names = useSelector(namesSelector);

  return (
    <ConfirmationModal isOpen={open} title="Suppression des gammes" toggle={toggle} confirm={deleteScopes}>
      <Typography>{`Confirmez-vous la suppression des gammes suivantes : ${names} ?`}</Typography>
    </ConfirmationModal>
  );
}

DeleteScopesConfirmationModal.propTypes = {
  open: bool,
  toggle: func,
};

DeleteScopesConfirmationModal.defaultProps = {
  open: false,
  toggle: Function.prototype,
};
