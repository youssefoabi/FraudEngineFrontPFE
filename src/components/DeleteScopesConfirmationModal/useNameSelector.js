import { path } from 'ramda';
import { useSelector } from 'react-redux';

export default function useNameSelector(id) {
  const selectName = useSelector(path(['entities', 'scopes', 'byId', id, 'name']));

  return { selectName };
}
