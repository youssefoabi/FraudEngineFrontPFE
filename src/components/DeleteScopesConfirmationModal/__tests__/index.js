import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import DeleteScopesConfirmationModal from '../';

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
  useSelector: () => ({}),
}));

describe(DeleteScopesConfirmationModal.name, () => {
  itRendersCorrectly(<DeleteScopesConfirmationModal />);
});
