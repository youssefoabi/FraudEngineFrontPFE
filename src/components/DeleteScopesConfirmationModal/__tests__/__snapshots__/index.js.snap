// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteScopesConfirmationModal renders correctly 1`] = `
<ConfirmationModal
  confirm={[Function]}
  isOpen={false}
  title="Suppression des gammes"
  toggle={[Function]}
>
  <WithStyles(ForwardRef(Typography))>
    Confirmez-vous la suppression des gammes suivantes : [object Object] ?
  </WithStyles(ForwardRef(Typography))>
</ConfirmationModal>
`;
